apps/swirl-docs/src/components/HeaderNavigation/index.tsx

```diff
@@ -1,14 +1,42 @@
 import Image from "next/image";
-import { useState } from "react";
 import icon from "@getflip/swirl-icons/icons/Close16.svg";
 import MobileNav from "./mobileNav";
 import Link from "next/link";
 import { navItems } from "@swirl/lib/navigation";
 import { useRouter } from "next/router";
+import { DesktopView, MobileView } from "../View/Views";
+import classNames from "classnames";
+import { useEffect, useState } from "react";
+
+export const HeaderLogo = () => {
+  return (
+    <Link href="/">
+      <a className="flex justify-center items-center mr-8">
+        <Image
+          alt="Swirl home"
+          src="/swirl-icon-temp.svg"
+          width={32}
+          height={32}
+        />
+        <span className="font-medium ml-3">Swirl</span>
+      </a>
+    </Link>
+  );
+};
 
 const HeaderNavigation = () => {
-  const [isMobileNavOpen, setIsMobileNavOpen] = useState(false);
   const router = useRouter();
+  const [isMobileNavOpen, setIsMobileNavOpen] = useState(false);
+
+  useEffect(() => {
+    if (typeof window) {
+      if (isMobileNavOpen) {
+        document.getElementsByTagName("body")[0].style.overflow = "hidden";
```

Comment:
**suggestion:** You could consider using https://github.com/willmcpo/body-scroll-lock here. It solves all the edge cases on mobile devices.

Reply:
Had no Idea a lib for that existed. I will take a look into it and implement it :)